Edit "Best practices for HTML in rich text" (#246)
CloudCannon/platform-documentation: articles/best-practices-for-rich-text.mdx

  https://cc-dam.imgix.net/documentation/images/default-thumbnail/2023-Q4/CloudCannon-Documentation-Thumbnail.png
description: 'Some HTML elements cannot be edited in the rich text editor. '
tags: []
related_articles: []
related_links: []
---
To get the most out of your editing experience, we recommend that you only include markup in your rich text files can be rendered and edited in CloudCannon.

CloudCannon uses the following definitions:

* Rich text content — any HTML or Markdown content you might want to edit in the Content Editor or a rich text input
* Custom markup — any HTML or Markdown structure that cannot be recreated in the rich text editor (e.g., `<div>` tags or class attributes). This is normally any structure that cannot be produced by the *WYSIWYG toolbar*.

To protect your content and your site, CloudCannon does not allow you to edit custom markup using the Content Editor or a rich text input. If you want to include custom markup in your rich text files, there are some limitations you should be aware of.

## Unsupported HTML elements in rich text

You cannot edit custom markup in rich text files using the Content Editor or a rich text input. This is because:

> * You can unintentionally delete custom markup from your files. As custom markup cannot be recreated in the rich text editor (i.e., it is invisible), it is easy to accidentally backspace or delete your HTML structures.

Reviewer: ```suggestion
* You can unintentionally delete custom markup from your files. As custom markup may be invisible in rich text editor, it is easy to accidentally backspace or delete your HTML structures, and you won't be able to recreate it without entering the source code editor.
```

Slight tweak maybe?

> * Some custom markup structure pose a risk for prevent cross-site scripting attacks (i.e., `<script>` tags) and other unwanted interactions between your code and the editing interface (i.e., `<style>` tags).

Reviewer: ```suggestion
* Some custom markup structures pose a risk for prevent cross-site scripting attacks (e.g.., `<script>` tags) and other unwanted interactions between your code and the editing interface (e.g., `<style>` tags).
```

not critical, but I think these should be e.g.s?
Also missing an 's'

> * Some custom markup structure pose a risk for prevent cross-site scripting attacks (i.e., `<script>` tags) and other unwanted interactions between your code and the editing interface (i.e., `<style>` tags).

Reviewer 2: ```suggestion
* Some custom markup structures pose a risk by enabling cross-site scripting attacks (i.e., `<script>` tags) and other unwanted interactions between your code and the editing interface (i.e., `<style>` tags).
```

Although you cannot edit your custom markup in the Content Editor or rich text inputs, CloudCannon will preserve it. Your custom markup will appear as an uneditable Snippet.

<comp.DocsImage path="https://cc-dam.imgix.net/documentation/images/editing/unsupported-html-rich-text/2024-Q1/CloudCannon-Documentation-Unsupported-HTML-Snippets.png" alt="Unsupported HTML in the rich text editor will appear as an uneditable Snippet." title="Uneditable Snippets in rich text" type="ui-snippet"/>

You can still edit the surrounding content in your files using a rich text editor. You can also edit your custom markup in the Source Editor.

### Best practice

> To prevent your files from becoming uneditable in the rich text editor, we recommend preparing your rich text files appropriately.

Reviewer: ```suggestion
To ensure your files remain editable in the rich text editor, we recommend the solutions below to avoid some common pitfalls:
```

Maybe not this exactly, but it might be nice to have a more positive perspective
